### ISSUE TITLE
How does this work?

### ISSUE THREAD

**Issue by Drflash55** [Apostolique/AposFeedingBot#2]
So, when you activate the AposFeedingBot, I see there's a status that can make you the Master, or the Slave...

Master works just like the regular, but with Slave, it only goes up to 50 mass, and then it just halts...
Is there supposed to be something else that goes with it?

**Comment by Apostolique** [comment]
There are many ways to join the same server. The easiest is to create a party link.

**Comment by Drflash55** [comment]
@Apostolique So, if I'm getting this right, when there's a 'master' on the server, any one who has the 'slave' status will try to find the master and then it gives itself to the 'master'...

I like this better, because Raga Minions don't take into consideration the dangers around them... Only thing is, when there's multiple bots, they combine into each other to stay alive...

Also, what if there's two masters on the server? How can you be a slave to only one master?

**Comment by Apostolique** [comment]
I can implement an ID system so that slaves can decided which master to follow.

**Comment by Drflash55** [comment]
@Apostolique That would be great...
That way, the bot doesn't get confused on which master to follow...
Would it be an interface on the ESC Menu, or does it have to be coded in the JS?

**Comment by Apostolique** [comment]
I could make an interface. If I do that though, I would need to delete masters when they are done. Right now, I update masters in a dirty way.

**Comment by Drflash55** [comment]
@Apostolique It can be a good thing if you make an interface for coding beginners, but I have tampered with some JS coding scripts, so all I'll have to do is just find the ID number that represents which master to follow, and then I'll be good.

Only thing is, you'll have to refresh the page to change the ID, while the interface can have real-time changes, which cuts time short.

**Comment by jasikanicolepi** [comment]
I don't know if Agar.io made some changes but I am unable to have 2 bots running under the same "party" game. If I have one bot in the game, and another attempt to join, it will cause one bot to refresh it's page, and the 2nd bot (the one that is trying to join) will then join the game.

**Comment by MasterPatch** [comment]
i see in some games multiply tiny names of people, they feed themselves, but there are like 20, i cant even get it to work! there is no tutorials or anything

**Comment by Apostolique** [comment]
I wrote the code only. How you get them all to join the same server is up to you. Using different browsers, proxies, computers, etc.

**Comment by GREEB** [comment]
I tried it for the first time...it's really good. I used it on an "Privat server" of some famous agar.io personality, if this is a thing :) anyways i went there because it's an ogar based server i think or they just don't care and let you connect up to like 20-30 times. I had bad connection. I used Facebook to get 50ish mass of spawn. Sometimes i tried to team with a bot but i had some major lag.

i made a video: https://youtu.be/b0MdkR5xbkM

How could i run the slaves so they use less cpu ( i7-4770K)?

Thanks for making this:+1:

**Comment by Apostolique** [comment]
That was cool to watch!

Right now there's no real way to make them use less cpu.

Perhaps https://github.com/FliiFe/Agar.io-bot-desktop-app will use less?

**Comment by GREEB** [comment]
Cool thanks will test

**Comment by Drflash55** [comment]
@Apostolique 
It would be great if it runs on a node-js script, so that it doesn't take up so much CPU usage by having to use multiple tabs to have it working...

**Comment by GREEB** [comment]
yeah...I'll try to do something

**Comment by makandz** [comment]
I'm going to see if I can manage to run this in a Visual Basic webcontrol. I can have this script implemented into around 10 or 15 Awesomium webcontrols and have all of them slaves and one master window for the master to play on. This could also use less ram because Awesomium isn't a ram hog from what I know, and it doesn't have to load any extra components (plugins, history, program itself, theme, or etc)

**Comment by Apostolique** [comment]
What is nice is that the bot itself is written almost as a standalone. Someone could write an alternate client than the launcher. One that doesn't even require a GUI and import the bot scripts how they want.

**Comment by makandz** [comment]
You're right. I'm just going to see what I can do with it for now, my JavaScript isn't the best :+1:

**Comment by CowCompanion** [comment]
So... I am really a noob at javascript, how do I run the program? :)

**Comment by henopied** [comment]
@Apostolique https://github.com/pulviscriptor/agario-client

**Comment by zakamanden** [comment]
But so if.i have more browsers open and on same server 1 master and A lot slave the shave will find and die to master ??

**Comment by zakamanden** [comment]
@henopied can you help me getting feedbot working

**Comment by Drflash55** [comment]
@zakamanden 
If you have more browsers open and have them set as the slaves, the slaves will behave as normal and they will head for you if you are set as the Master, but you damn better have a really good processor and some decent RAM (talking about 6GB).

Also, Agar.io doesn't allow that. You can only do this by going to an Ogar Private Server. The only way to get multiple bots hosted by you on a party server is to set up a proxy for each bot, and that takes a lot of effort...

**Comment by AgarioCheats** [comment]
Apostolique, you're talented. How come you're not doing this for a living? We can cooperate if you wish to!

**Comment by Apostolique** [comment]
What did you have in mind?

I'm not doing this for a living since I'm still a university student. The projects that I work on give me valuable experience though.

Thanks btw!
